processing/src/main/java/io/druid/segment/SingleScanTimeDimSelector.java

```diff
@@ -74,6 +80,12 @@ public boolean matches()
       {
         return Objects.equals(lookupName(getDimensionValueIndex()), value);
       }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        inspector.visit("selector", SingleScanTimeDimSelector.this);
```

Comment:
`SingleScanTimeDimSelector.this` inspects `extractionFn`. `timeValues` is always `ArrayList`, shouldn't be inspected.